compiler/riscv/codegen/ControlFlowEvaluator.cpp

```diff
@@ -336,89 +338,103 @@ OMR::RV::TreeEvaluator::icmpneEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 
    }
 
+// also handles bcmplt and scmplt
 TR::Register *
 OMR::RV::TreeEvaluator::icmpltEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
    return icmpHelper(TR::InstOpCode::_slt, TR::InstOpCode::bad, 0, node, false, cg);
    }
 
+// also handles bcmple and scmple
 TR::Register *
 OMR::RV::TreeEvaluator::icmpleEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
    return icmpHelper(TR::InstOpCode::_slt, TR::InstOpCode::_sltiu, 1, node, true, cg);
    }
 
+// also handles bcmpge and scmpge
 TR::Register *
 OMR::RV::TreeEvaluator::icmpgeEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
    return icmpHelper(TR::InstOpCode::_slt, TR::InstOpCode::_xori, 1, node, false, cg);
    }
 
+// also handles bcmpgt and scmpgt
 TR::Register *
 OMR::RV::TreeEvaluator::icmpgtEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
    return icmpHelper(TR::InstOpCode::_slt, TR::InstOpCode::bad, 0, node, true, cg);
    }
 
+// also handles bucmplt and sucmplt
```

Comment:
I don't think reusing the existing `iucmp*` evaluators for `bucmp*` and `sucmp*` will work without modifications, since the registers containing the values of the child nodes are always sign-extended and need to be zero-extended before they can be compared. That being said, this does expose a pretty glaring blind spot in our testing that probably needs to be addressed...

Reply:
You're right, thanks! 

As for the testing issues - yes, I noticed that some time ago. I have done some work to address this, perhaps I should finish that one first. This would also allow to push some other PR that contains `zeroExtend()` helper which can be reused here as well...